File: 프로그래머스/0/120913. 잘라서 배열로 저장하기/잘라서 배열로 저장하기.js
function solution(my_str, n) {
  const arr = my_str.split('')
  for (let i = n; i < arr.length; i += n+1) {
    arr.splice(i, 0, ',')
    if(i >= arr.length) break
  }
  return arr.join('').split(',')
}

File: 프로그래머스/0/120913. 잘라서 배열로 저장하기/잘라서 배열로 저장하기.java
import java.util.ArrayList;

class Solution {
    public String[] solution(String my_str, int n) {
        ArrayList<String> answer = new ArrayList<>();
        
        for(int i = 0; i < my_str.length(); i += n) {
            answer.add(my_str.substring(i, Math.min(i + n, my_str.length())));
        }
        return answer.toArray(new String[0]);
    }
}